python/az/aro/azext_aro/_aad.py

```diff
@@ -29,15 +30,15 @@ def __init__(self, cli_ctx):
             credentials, tenant_id, base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
         configure_common_settings(cli_ctx, self.client)
 
-    def create_application(self, display_name):
+    def create_application(self, display_name, cli_ctx):
```

Comment:
You are not fixing calls of this anywhere? 
`app, client_secret = aad.create_application(cluster_resource_group or 'aro-' + random_id)`

Did it works locally for you?

Reply:
Or Im missing python magic here?

Reply:
+1. I think `python/az/aro/azext_aro/custom.py` also needs to be updated if we go the route of passing `cli_ctx` into method calls. Also please make it first argument for consistency with other methods/functions.

But the question is: do we actually need to do this? Don't we already pass `cli_ctx` into constructor above?

```python
    def __init__(self, cli_ctx):
        profile = Profile(cli_ctx=cli_ctx)
        # ...
```

Reply:
Changes made to reference cli_ctx as an instance variable so we do not need to pass it from custom.py. Was able to test this by successfully creating a cluster in FairFax INT using local az cli extension built with code in this PR. Awaiting further review and feedback :) @m1kola @mjudeikis